### PR TITLE
Access Token Response expires_in should be a JSON number

### PR DESCRIPTION
**Pull request by roll57**
Closes gh-9685

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@roll57 Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security&pullRequestId=9723)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-security&pullRequestId=9723) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@roll57 Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security&pullRequestId=9723)!

File: oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/endpoint/OAuth2AccessTokenResponseMapConverter.java [hunk 1]
```diff
@@ -34,14 +34,14 @@
  * @since 5.3
  */
 public final class OAuth2AccessTokenResponseMapConverter
-		implements Converter<OAuth2AccessTokenResponse, Map<String, String>> {
+		implements Converter<OAuth2AccessTokenResponse, Map<String, Object>> {
```
Comment (jgrandja) [inline oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/endpoint/OAuth2AccessTokenResponseMapConverter.java:37]: This change will break existing applications.

**Comment by jgrandja** [comment]
Thanks for the PR @roll57 but the changes in this PR would break existing applications. Please see [comment](https://github.com/spring-projects/spring-security/issues/9685#issuecomment-837137111).

Also, I'm planning on providing the fix for gh-9685 as I originally assigned it to me. It's always best to reach out first before you start work on an issue so we can let you know if it's available or not.

I'm going to close this PR based on my explanation as per above.

**Pull request closed without merging**